Rename all occurences of "accelerator" into "gpu"

Krebrov001 opened the pull request:
- Relates to #2171 
- Fixes #2172 

Rename all occurences of "accelerator" into "gpu" within files in the `MANIFEST`.
Also rename the actual files themselves:

```
service/src/AcceleratorTopo.cpp
service/src/AcceleratorTopo.hpp
service/src/AcceleratorTopoNull.cpp
service/src/AcceleratorTopoNull.hpp
service/src/LevelZeroAcceleratorTopo.cpp
service/src/LevelZeroAcceleratorTopo.hpp
service/src/NVMLAcceleratorTopo.cpp
service/src/NVMLAcceleratorTopo.hpp
service/test/AcceleratorTopoNullTest.cpp
service/test/LevelZeroAcceleratorTopoTest.cpp
service/test/MockAcceleratorTopo.hpp
service/test/NVMLAcceleratorTopoTest.cpp
```

cmcantalupo commented:
accelerator -> gpu
Accelerator -> GPU
ACCELERATOR -> GPU

lhlawson commented:
- [ ] **Intel GPU testing is still required.** 
- [x] NVIDIA GPU spot testing on an HPE SX40 found no issues. 
> HPE-SX40 basic reads return as expected, along with valid errors as expected for domain_idx out of bound (reading a gpu that doesn't exist) and unsupported domains (gpu_chip for this system):
> ```
> [hpe-sx40:~]geopmread GPU_MEMORY_ACTIVITY gpu 0
> 1.727236320129163e-08
> [hpe-sx40:~]geopmread GPU_COMPUTE_ACTIVITY gpu 0
> 0
> [hpe-sx40:~]geopmread GPU_MEMORY_ACTIVITY gpu 0
> 5.820883204884875e-09
> [hpe-sx40:~]geopmread GPU_POWER gpu 0
> 46.175
> [hpe-sx40:~]geopmread GPU_POWER board 0
> 177.898
> [hpe-sx40:~]geopmread GPU_POWER gpu 15
> Error: cannot read signal: <geopm> Invalid argument: PlatformIOImp::read_signal(): domain_idx is out of range: at src/PlatformIO.cpp:482
> [hpe-sx40:~]geopmread GPU_POWER gpu 3
> 43.754
> [hpe-sx40:~]geopmwrite GPU_FREQUENCY_CONTROL board 0 1530000000
> [hpe-sx40:~]geopmread GPU_FREQUENCY_STATUS board 0
> 1530000000
> [hpe-sx40:~]geopmwrite GPU_FREQUENCY_CONTROL gpu 0 1530000000
> [hpe-sx40:~]geopmwrite GPU_FREQUENCY_CONTROL gpu 0 135000000
> [hpe-sx40:~]geopmread GPU_FREQUENCY_STATUS gpu 0
> 135000000
> [hpe-sx40:~]geopmread GPU_FREQUENCY_STATUS board 0
> 1181250000
> [hpe-sx40:~]geopmread GPU_FREQUENCY_STATUS gpu_chip 0
> Error: cannot read signal: <geopm> Invalid argument: PlatformIOImp::read_signal(): domain 10 is not valid for signal "GPU_FREQUENCY_STATUS": at src/PlatformIO.cpp:543
> [hpe-sx40:~]geopmwrite NVML::GPU_FREQUENCY_RESET_CONTROL board 0 1
> [hpe-sx40:~]geopmread GPU_FREQUENCY_STATUS board 0
> 135000000
> ```
> 
> Testing with a basic PARRES dgemm run shows reports working:
> ```
>     Application Totals:
>       runtime (s): 15.2876
> <removed non-gpu readings as uninteresting to debug>
>       gpu-energy (J): 12835.6
>       gpu-power (W): 840.296
>       gpu-frequency (Hz): 1.43566e+09
> ```
> 
> Similarly tracing of GPU workloads appears to be unaffected by the change, beyond signal domain changes:
> ![image](https://user-images.githubusercontent.com/23505009/167680381-5887ea8a-ae3e-42a3-ac78-710b5eff42dc.png)
> 
> Not tested for V100s: 
> - non-root privileges 
> - all GPUs, beyond testing of board level readings
> - accuracy of readings, beyond visual inspection vs previous readings on the same system.

Krebrov001 committed:
Performed mass rename operation

The replacements are according to the following rules:

```
BOARD_NIC -> NIC
board_nic -> nic
PACKAGE_NIC -> PACKAGE_INTEGRATED_NIC
package_nic -> package_integrated_nic

BOARD_MEMORY -> MEMORY
board_memory -> memory
PACKAGE_MEMORY -> PACKAGE_INTEGRATED_MEMORY
package_memory -> package_integrated_memory

BOARD_GPU -> GPU
board_gpu -> gpu
PACKAGE_GPU -> PACKAGE_INTEGRATED_GPU
package_gpu -> package_integrated_gpu

BOARD_ACCELERATOR -> GPU
board_accelerator -> gpu
PACKAGE_ACCELERATOR -> PACKAGE_INTEGRATED_GPU
package_accelerator -> package_integrated_gpu

accelerator -> gpu
Accelerator -> GPU
ACCELERATOR -> GPU
aCCELERATOR -> GPU

accel -> gpu
ACCEL -> GPU
Accel -> GPU
aCCEL -> GPU
```

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 committed:
Renamed files to use the word `GPU` instead of the word `Accelerator`

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>

Krebrov001 commented:
Please review
@cmcantalupo @bgeltz  @lhlawson

bgeltz commented on service/docs/source/security.rst:
```diff
@@ -283,7 +283,7 @@ LevelZero sysman library interface
 
 The LevelZero sysman library interface enables users to monitor and
-control Intel accelerator devices. These signals and controls include
+control Intel gpu devices. These signals and controls include
```
```suggestion
control Intel GPU devices. These signals and controls include
```

bgeltz commented on service/configure.ac:
```diff
@@ -97,7 +97,7 @@ AC_SUBST([enable_levelzero])
 
 AC_ARG_ENABLE([nvml],
-  [AS_HELP_STRING([--enable-nvml], [Enables use of the NVML library to support NVML board accelerators])],
+  [AS_HELP_STRING([--enable-nvml], [Enables use of the NVML library to support NVML board gpus])],
```
```suggestion
  [AS_HELP_STRING([--enable-nvml], [Enables use of the NVML library to support NVML board GPUs])],
```

bgeltz commented on service/configure.ac:
```diff
@@ -110,7 +110,7 @@ AC_SUBST([enable_nvml])
 
 AC_ARG_ENABLE([dcgm],
-  [AS_HELP_STRING([--enable-dcgm], [Enables use of the DCGM library to support NVIDIA board accelerators])],
+  [AS_HELP_STRING([--enable-dcgm], [Enables use of the DCGM library to support NVIDIA board gpus])],
```
```suggestion
  [AS_HELP_STRING([--enable-dcgm], [Enables use of the DCGM library to support NVIDIA board GPUs])],
```

bgeltz commented on service/configure.ac:
```diff
@@ -150,17 +150,17 @@ AC_SUBST([enable_coverage])
 if test "x$enable_dcgm" = "x1" ; then
   ENABLE_DCGM=True
-  AC_DEFINE([GEOPM_ENABLE_DCGM], [ ], [Enables use of the DCGM library to support NVIDA board accelerators])
+  AC_DEFINE([GEOPM_ENABLE_DCGM], [ ], [Enables use of the DCGM library to support NVIDA board gpus])
```
```suggestion
  AC_DEFINE([GEOPM_ENABLE_DCGM], [ ], [Enables use of the DCGM library to support NVIDA board GPUs])
```

bgeltz commented on service/configure.ac:
```diff
@@ -150,17 +150,17 @@ AC_SUBST([enable_coverage])
 if test "x$enable_nvml" = "x1" ; then
   ENABLE_NVML=True
-  AC_DEFINE([GEOPM_ENABLE_NVML], [ ], [Enables use of the NVML library to support NVML board accelerators])
+  AC_DEFINE([GEOPM_ENABLE_NVML], [ ], [Enables use of the NVML library to support NVML board gpus])
```
```suggestion
  AC_DEFINE([GEOPM_ENABLE_NVML], [ ], [Enables use of the NVML library to support NVML board GPUs])
```

bgeltz commented on service/configure.ac:
```diff
@@ -150,17 +150,17 @@ AC_SUBST([enable_coverage])
     if test "x$enable_levelzero" = "x1" ; then
         ENABLE_LEVELZERO=True
-        AC_DEFINE([GEOPM_ENABLE_LEVELZERO], [ ], [Enables use of the levelzero library to support levelzero board accelerators])
+        AC_DEFINE([GEOPM_ENABLE_LEVELZERO], [ ], [Enables use of the levelzero library to support levelzero board gpus])
```
```suggestion
        AC_DEFINE([GEOPM_ENABLE_LEVELZERO], [ ], [Enables use of the levelzero library to support levelzero board GPUs])
```

bgeltz commented on service/src/DCGMDevicePool.hpp:
```diff
@@ -38,29 +38,29 @@ namespace geopm
 
-            /// @brief Number of accelerators that support DCGM on the platform.
-            /// @return Number of accelerators supported by DCGM.
+            /// @brief Number of gpus that support DCGM on the platform.
```
```suggestion
            /// @brief Number of GPUs that support DCGM on the platform.
```

bgeltz commented on service/src/DCGMDevicePool.hpp:
```diff
@@ -38,29 +38,29 @@ namespace geopm
-            /// @param [in] accel_idx The index indicating a particular
-            ///        accelerator.
+            /// @param [in] gpu_idx The index indicating a particular
+            ///        gpu.
```
```suggestion
            ///        GPU.
```

bgeltz commented on service/src/DCGMDevicePool.hpp:
```diff
@@ -38,29 +38,29 @@ namespace geopm
+            virtual double sample(int gpu_idx, int field_id) const = 0;
 
-            /// @brief Query DCGM for the latest value for an accelerator.
+            /// @brief Query DCGM for the latest value for an gpu.
```
```suggestion
            /// @brief Query DCGM for the latest value for an GPU.
```

bgeltz commented on service/src/DCGMDevicePool.hpp:
```diff
@@ -38,29 +38,29 @@ namespace geopm
-            ///        accelerator.
-            virtual void update(int accel_idx) = 0;
+            /// @param [in] gpu_idx The index indicating a particular
+            ///        gpu.
```
```suggestion
            ///        GPU.
```

bgeltz commented on service/src/GPUTopo.hpp:
```diff
@@ -3,32 +3,32 @@
-            /// @brief Number of accelerators on the platform.
+            GPUTopo() = default;
+            virtual ~GPUTopo() = default;
+            /// @brief Number of gpus on the platform.
```
```suggestion
            /// @brief Number of GPUs on the platform.
```

bgeltz commented on service/src/GPUTopo.hpp:
```diff
@@ -3,32 +3,32 @@
-            /// @brief CPU Affinitization set for a particular accelerator
+            virtual int num_gpu(void) const = 0;
+            virtual int num_gpu(int domain) const = 0;
+            /// @brief CPU Affinitization set for a particular gpu
```
```suggestion
            /// @brief CPU Affinitization set for a particular GPU
```

bgeltz commented on service/src/GPUTopo.hpp:
```diff
@@ -3,32 +3,32 @@
             /// @param [in] domain The GEOPM domain type
             /// @param [in] domain_idx The index indicating a particular
-            ///        accelerator
+            ///        gpu
```
```suggestion
            ///        GPU
```

bgeltz commented on service/src/LevelZero.cpp:
```diff
@@ -350,24 +350,24 @@ namespace geopm
-        //  TODO: this should be expanded to return all supported accel types.
-        //  Right now that is only board_gpus
-        return num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR);
+        //  TODO: this should be expanded to return all supported gpu types.
+        //  Right now that is only gpus
```
```suggestion
        //  TODO: this should be expanded to return all supported GPU types.
        //  Right now that is only GPUs
```

bgeltz commented on service/src/LevelZero.cpp:
```diff
@@ -350,24 +350,24 @@ namespace geopm
     {
-        //  TODO: this should be expanded to return all supported accel types.
-        //  Right now that is only board_gpus
+        //  TODO: this should be expanded to return all supported gpu types.
+        //  Right now that is only gpus
```
```suggestion
        //  TODO: this should be expanded to return all supported GPU types.
        //  Right now that is only GPUs
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
-            /// @brief Number of accelerators on the platform.
-            /// @return Number of LevelZero accelerators.
-            virtual int num_accelerator() const = 0;
+            /// @brief Number of gpus on the platform.
+            /// @return Number of LevelZero gpus.
```
```suggestion
            /// @brief Number of GPUs on the platform.
            /// @return Number of LevelZero GPUs.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
+            virtual int num_gpu() const = 0;
 
-            /// @brief Number of accelerators on the platform.
+            /// @brief Number of gpus on the platform.
```
```suggestion
            /// @brief Number of GPUs on the platform.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] domain The GEOPM domain type being targeted
-            /// @return Number of LevelZero accelerators or accelerator chips.
-            virtual int num_accelerator(int domain) const = 0;
+            /// @return Number of LevelZero gpus or gpu chips.
```
```suggestion
            /// @return Number of LevelZero GPUs or GPU chips.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @brief Get the LevelZero device actual frequency in MHz
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_domain_idx The LevelZero index indicating a particular
-            ///        domain of the accelerator.
-            /// @return Accelerator device core clock rate in MHz.
+            ///        domain of the gpu.
```
```suggestion
            ///        domain of the GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @brief Get the LevelZero device mininmum frequency in MHz
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_domain_idx The LevelZero index indicating a particular
-            ///        domain of the accelerator.
-            /// @return Accelerator minimum frequency in MHz.
+            ///        domain of the gpu.
```
```suggestion
            ///        domain of the GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @brief Get the LevelZero device maximum frequency in MHz
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_domain_idx The LevelZero index indicating a particular
-            ///        domain of the accelerator.
-            /// @return Accelerator maximum frequency in MHz.
+            ///        domain of the gpu.
```
```suggestion
            ///        domain of the GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             ///        control range in MHz
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_domain_idx The LevelZero index indicating a particular
-            ///        domain of the accelerator.
-            /// @return Accelerator minimum and maximum frequency range in MHz.
+            ///        domain of the gpu.
```
```suggestion
            ///        domain of the GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @brief Get the LevelZero device active time and timestamp in microseconds
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @brief Get the LevelZero device active time in microseconds
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_domain_idx The LevelZero index indicating a particular
-            ///        domain of the accelerator.
-            /// @return Accelerator active time in microseconds.
+            ///        domain of the gpu.
```
```suggestion
            ///        domain of the GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             ///        active time value in microseconds
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_domain_idx The LevelZero index indicating a particular
-            ///        domain of the accelerator.
-            /// @return Accelerator device timestamp for the active time value in microseconds.
+            ///        domain of the gpu.
```
```suggestion
            ///        domain of the GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
-            /// @return Accelerator default power limit in milliwatts
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
-            /// @return Accelerator minimum power limit in milliwatts
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
-            /// @return Accelerator maximum power limit in milliwatts
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
-            /// @return Accelerator energy in microjoules and timestamp in microseconds
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
-            /// @return Accelerator energy in microjoules.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
-            /// @return Accelerator energy timestamp in microseconds
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

bgeltz commented on service/src/LevelZero.hpp:
```diff
@@ -26,129 +26,129 @@ namespace geopm
             /// @brief Set min and max frequency for LevelZero device.
             /// @param [in] l0_device_idx The index indicating a particular
-            ///        Level Zero accelerator.
+            ///        Level Zero gpu.
```
```suggestion
            ///        Level Zero GPU.
```

Krebrov001 commented:
I removed the `rename_accelerator.py` from this branch/pull request `issue-2171`.

It is now located on the branch `rename_script`.

bgeltz reviewed (changes requested):
We should be consistent about when GPU is capitalized in comments and error strings.  When it's used in a sentence or similar, it should always be GPU or GPUs.

I marked many of these, but I did not mark them all.  There are many more that need to be corrected.

Krebrov001 committed:
Renamed all occurences of `gpu` to `GPU` and `cpu` to `CPU`

- in comments and string literals
- variables, command line options, configuration and settings stay the same

Signed-off-by: Konstantin Rebrov <konstantin.rebrov@intel.com>
